src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxePsiFieldImpl.java

```diff
@@ -213,4 +208,16 @@ public boolean hasModifierProperty(@HaxePsiModifier.ModifierConstant @NonNls @No
     return this.getModifierList().hasModifierProperty(name);
   }
 
+  @NotNull
+  @Override
+  public SearchScope getUseScope() {
+    final PsiElement localVar = UsefulPsiTreeUtil.getParentOfType(this, HaxeLocalVarDeclaration.class);
+    if(localVar != null) {
+      final PsiElement outerBlock = UsefulPsiTreeUtil.getParentOfType(localVar, HaxeBlockStatement.class);
```

Comment:
That is a good fix.  However, the point of the sample was really to point out that local scopes (such as functions) aren't *always* surrounded by HaxeBlockStatements.  In my version of the example, the variable 'a' inside of the for loop hides the variable with the same name and different type declared at the class level.  Therefore, locating the scope requires searching for more than an enclosing block statement; it requires looking for enclosing function definitions, too.